sdcm/nemesis.py

```diff
@@ -2698,6 +2697,64 @@ def disrupt_mgmt_backup_specific_keyspaces(self):
     def disrupt_mgmt_backup(self):
         self._mgmt_backup(backup_specific_tables=False)
 
+    def disrupt_mgmt_restore(self):
+        def get_total_scylla_partition_size():
+            result = self.cluster.nodes[0].remoter.run("df -k | grep /var/lib/scylla")  # Size in KB
+            free_space_size = int(result.stdout.split()[1]) / 1024 ** 2  # Converting to GB
+            return free_space_size
+
+        def choose_snapshot(snapshots_dict):
+            snapshot_groups_by_size = snapshots_dict["snapshots"]
+            total_partition_size = get_total_scylla_partition_size()
+            all_snapshot_sizes = sorted(list(snapshot_groups_by_size.keys()), reverse=True)
+            fitting_snapshot_sizes = [size for size in all_snapshot_sizes if total_partition_size / size >= 20]
+            if self.tester.test_duration < 1000:
```

Comment:
It's a bit arbitrary, mostly because the 100gb restore could take quite a while. The restore itself could take up to 2 hours, and the stress can take quite a while to. So, to avoid having runs where the only executed nemesis is the restore one, I decided 1000 minutes would be a good limit.

Reply:
inside of each snapshot on `defaults/manager_persistent_snapshots.yaml` you have a `expected_timeout`... why not to use it instead of arbitrary wait 1000 minutes?

Reply:
This `expected timeout` is only used with regards to the task, not the nemesis as a whole.

Reply:
but the task should be shorter than the nemesis itself, no?
otherwise we can get into a situation where the nemesis timed out, but the `sctool` task not

Reply:
task duration is in minutes
expected_timeout is in seconds (stated in the yaml file)

Reply:
I still didn't get an answer where the 1000 min is coming from, and how do we know it won't take more than that ?

I.e. what is time depended on ? The size of data to restore ? To load on the cluster ? both ?

I rather see usage of adaptive_timeout, so we can start collecting the timing information of such operations, and later create formulas, instead of just putting too big timeouts.